paimon-core/src/main/java/org/apache/paimon/operation/KeyValueFileStoreScan.java

```diff
@@ -121,10 +134,33 @@ protected boolean filterByStats(ManifestEntry entry) {
         }
 
         return filter.test(
```

Comment:
Add "long schemaId = entry.file().schemaId();" in KeyValueFileStoreScan#filterByStats, and then use schemaId in KeyValueFileStoreScan#filterByStats and KeyValueFileStoreScan#testFileIndex.